src/main/java/de/tum/in/www1/artemis/domain/Complaint.java

```diff
@@ -154,18 +154,14 @@ public void setComplaintResponse(ComplaintResponse complaintResponse) {
      * @param participant either a team or user
      */
     public void setParticipant(Participant participant) {
-        if (participant instanceof User) {
-            this.student = (User) participant;
-        }
-        else if (participant instanceof Team) {
-            this.team = (Team) participant;
-        }
-        else if (participant == null) {
-            this.student = null;
-            this.team = null;
-        }
-        else {
-            throw new Error("Unknown participant type");
+        switch (participant) {
+            case User user -> this.student = user;
+            case Team team1 -> this.team = team1;
+            case null -> {
+                this.student = null;
+                this.team = null;
+            }
+            default -> throw new Error("Unknown participant type");
```

Comment:
Is it possible to seal the participant in order to remove the default case?

Reply:
Ah sorry, I'm sick rn so I'm not too active on my computer hence the delay.

What exactly do you mean by "seal"? Do you mean the Java 17 `sealed` keyword and that I should make `Participant` a `sealed` class?

Reply:
No problem at all. Yes, I mean "sealed". I think the pattern matching should detect that you have all cases. I'm not 100% sure whether the base is an interface, abstract class or normal class, but you have some possibilities with sealed here: https://www.freecodecamp.org/news/write-better-java-code-pattern-matching-sealed-classes/

Reply:
![image](https://github.com/user-attachments/assets/ac245cee-5957-476a-aca4-581e4d4478ee)

Reply:
Oh yeah, that looks really nice, thx for the suggestion.
I'll look more into `sealed`. I've looked at it on Java 17 release, but haven't ever actually used it in a project since, so I just want to make sure that by adding this, I don't accidently "destroy" something else (be it compatibility, performance, etc.).

I'll push the changes tomorrow (if my research didn't reveal something bad).

Reply:
The problem could be the package structure of Artemis here. Maybe that's something to discuss with @krusche from a programming perspective, it would be really good to use sealed here because it ensures that you'll find all places if another type of participant occurs. You are also welcome to contact me via Slack in this regard.

Reply:
Yeah, the package structure is a problem.

I tried a few things, but we'd basically have to move `de.tum.in.www1.artemis.domain.User.java` and `de.tum.in.www1.artemis.domain.Team.java` into the `de.tum.in.www1.artemis.domain.participation` folder, otherwise it complains about `Class is not allowed to extend sealed class from another package`.

But yeah, in order to address that we'd definitely need to talk to @krusche to ensure we even can make such changes.

Reply:
Errrr. I also just realized that since this PR had been created some time ago (but I couldn't work on it cause vacation, into retake preparation into sickness) the entire file structure changed.

Like, it's no longer `de.tum.in.www1.artemis`, but now it's `de.tum.cit.aet.artemis`.

So yeah... I guess I'll need to spend some time again to manually apply all the changes to the new files, cause I honestly have no idea how to automatically merge this, if it's even possible.

Reply:
This also means that the locations I said earlier, i.e. 

> we'd basically have to move `de.tum.in.www1.artemis.domain.User.java` and `de.tum.in.www1.artemis.domain.Team.java` into the `de.tum.in.www1.artemis.domain.participation` folder

should've said this:

> we'd basically have to move `de.tum.cit.aet.artemis.core.domain.User.java` and `de.tum.cit.aet.artemis.exercise.domain.Team.java` into the `de.tum.cit.aet.artemis.exercise.domain.participation` folder